### DISCUSSION TITLE
start at login, nativeApplication.startAtLogin seems to not work on os x

### DISCUSSION THREAD

**Discussion by gontis** [airsdk/Adobe-Runtime-Support#1609] [Guides & Tutorials]
if (NativeApplication.supportsStartAtLogin) {
	NativeApplication.nativeApplication.startAtLogin = val
}

macOs, high sierra
setting this has no effect.
at this point I am not sure if this is my build problem or AIR bug.
anyone could maybe direct me how its implemented under the hood, to what plist or launch agent it writes itself?

thank you guys.

**Comment by itlancer** [comment]
This is AIR bug: https://github.com/airsdk/Adobe-Runtime-Support/issues/861

**Reply by gontis** [reply]
so if I use older, lets say, 32 air sdk it should work?

**Reply by ajwfrost** [reply]
Yes I believe it would work with AIR 32: on Windows we traced this to a problem with the certificate validation code that we'd not originally received from Adobe, so it would have been okay with their implementation i.e. AIR 32. On macOS we can double-check but it's likely a similar story that's preventing it from being set up properly.

thanks

**Reply by gontis** [reply]
..using really old 20.00 AIr for desktop build on mac .. still no dice. using self signed cert, might that be an issue?

**Reply by ajwfrost** [reply]
Hmmm... interestingly the mac code looks a lot simpler but is doing a number of OS calls to get this set up. It thinks that it's worked (a call to `CFPreferencesSynchronize` returns 'true') otherwise you'd get that same exception that's thrown on Windows.

We can look into this one a little more...

thanks

**Reply by ajwfrost** [reply]
... a quick check through the documentation and it might be that the current implementation has stopped working (across all AIR versions) after a macOS update, i.e. it's the OS blocking this now, as the app will be running in a sandboxed environment which seems to mean that the calls to set up the launch preferences aren't doing anything.

So we will need to create a new 'helper' for this. Seems somewhat convoluted but we need to have a helper process within the app itself which we can then get registered with the system and this is then launched and can decide to launch the app. I'm thinking that this may only work if it's a 'bundle' application rather than something that relies upon the shared AIR runtime, can I check how you're deploying your app?

thanks

**Reply by gontis** [reply]
thanks for your time!  might be the case. when did that OS update could have happened? we have tried various configurations. what was driving me crazy, is what I 100% have tested this ability at the beginning of developing and now, just before release it wont work. 

runtime embedded would be best, so I guess it can be bundled, right?

 In case anyone else is having similar issue, I have managed to get this working on windows with installer.

don't know if it will be of any use for you:
on macOS ~/Library/Preferences/loginwindow.plist
key gets created with app upon setting NativeApplication.nativeApplication.startAtLogin = true
however, it does not seem to affect actual launch

much appreciated, Aivaras

**Reply by ajwfrost** [reply]
Looking at some of the documentation it might be that this was a change in mojave (10.14) although that seems quite a while ago now..?! but yes I think it's related to the app sandbox approach so the key may be created on the OS but isn't enough to get the OS to launch it....

**Comment by gontis** [comment]
strange.

I am testing on High Sierra 10.13. 6
I am 99% sure I was able to get that functionality a few months ago
(setting app to launch on login) Now it just fails. could it have been a
silent update from Apple?
or accessing that system variable(s) from a testing environment with
several versions of the same app somehow broke it?
It's very opaque. I will try to run it on a freshly reinstalled system to
see how it goes.
Meanwhile, launch is due and we need this functionality, any ideas how to
hack it in somehow?
Same on windows - whilst it does work with installer, standalone version
would be preferred.
thanks for your time, looking forward, a
